### ISSUE TITLE
Should we sink(2) ?

### ISSUE THREAD

**Issue by franciscotln** [Andarist/callbag-take-while#1]
I was wondering if we really should send 2 downwards after the predicate is not satisfied. IMO there should be only a termination of the source `sourceTalkback(2)` and not sending `"completed"` to the sink because the source is not necessarily completed, it was just terminated.

Moreover: from what I understood from the [spec](https://github.com/callbag/callbag):
> After the handshake, the source MAY terminate the sink. Alternatively, the sink MAY terminate the source after the handshake has occurred. If the source terminates the sink, then the sink SHOULD NOT terminate the source, and vice-versa. In other words, termination SHOULD NOT be mutual.

This should not be done :-) 
What do you think?

**Comment by Andarist** [comment]
Hm, the spec describes a simple source-sink relationship and I believe this operator conforms to spec - it's the operator which sends 2 downwards & upwards "at the same time", source will think that sink has unsubscribed and the sink will think that source has been completed (and spec says that they shouldnt send 2 in the other direction then).

This operator is in its core really similar to `take`, it only uses a predicate function to determine when to end the communication instead of internal counter. And when `take` reaches its limit it sends 2 in both directions - https://github.com/staltz/callbag-take/blob/8046c77574e1c52d374a90f1af371a8c920dda4f/index.js#L16-L19

Ofc I'm happy to discuss this further but atm I believe the implementation here is correct (at least in this regard)

**Comment by franciscotln** [comment]
yes I think you're right :) thanks for taking your time anyway

**Comment by Andarist** [comment]
No problem! Always happy to discuss things.
